x-pack/legacy/plugins/ml/server/models/data_recognizer/data_recognizer.js

```diff
@@ -811,45 +811,56 @@ export class DataRecognizer {
   }
 
   applyJobConfigOverrides(moduleConfig, jobOverrides, jobPrefix = '') {
-    if(jobOverrides !== undefined && jobOverrides !== null) {
-      if (typeof jobOverrides !== 'object') {
-        throw Boom.badRequest(
-          `Incompatible jobOverrides type (${typeof jobOverrides}). It needs to be an object or array of objects.`
-        );
-      }
+    if (jobOverrides === undefined || jobOverrides === null) {
+      return;
+    }
 
-      // jobOverrides could be a single object or an array of objects.
-      // if single, convert to an array
-      const overrides = Array.isArray(jobOverrides) ? jobOverrides : [jobOverrides];
-      const { jobs } = moduleConfig;
+    if (typeof jobOverrides !== 'object') {
+      throw Boom.badRequest(
+        `Incompatible jobOverrides type (${typeof jobOverrides}). It needs to be an object or array of objects.`
+      );
+    }
 
-      // separate all the overrides.
-      // the overrides which don't contain a job id will be applied to all jobs in the module
-      const generalOverrides = [];
-      const jobSpecificOverrides = [];
-      overrides.forEach(o => {
-        if (o.job_id === undefined) {
-          generalOverrides.push(o);
-        } else {
-          jobSpecificOverrides.push(o);
-        }
-      });
+    // jobOverrides could be a single object or an array of objects.
+    // if single, convert to an array
+    const overrides = Array.isArray(jobOverrides) ? jobOverrides : [jobOverrides];
+    const { jobs } = moduleConfig;
 
-      generalOverrides.forEach(o => {
-        jobs.forEach(({ config }) => merge(config, o));
-      });
+    // separate all the overrides.
+    // the overrides which don't contain a job id will be applied to all jobs in the module
+    const generalOverrides = [];
+    const jobSpecificOverrides = [];
 
-      jobSpecificOverrides.forEach(o => {
-        // for each override, find the relevant job.
-        // note, the job id already has the prefix prepended to it
-        const job = jobs.find(j => j.id === `${jobPrefix}${o.job_id}`);
-        if (job !== undefined) {
-          // delete the job_id in the override as this shouldn't be overridden
-          delete o.job_id;
-          merge(job.config, o);
-        }
+    overrides.forEach(override => {
+      if (override.job_id === undefined) {
+        generalOverrides.push(override);
+      } else {
+        jobSpecificOverrides.push(override);
+      }
+    });
+
+    generalOverrides.forEach(generalOverride => {
+      jobs.forEach(job => {
+        job.config = {
+          ...job.config,
```

Comment:
`merge` was originally used here to allow overriding items deep in the job or datafeed config.
for example, a user should be able to just override the bucket span by supplying
```
{
"analysis_config": {
    "bucket_span": "1h"
  }
}

```

Reply:
a possible solution here would be to revert back to  `merge` and then to explicitly override `groups`, `influencers` and `query` if they have been supplied.

Reply:
Fixed in https://github.com/elastic/kibana/pull/49310/commits/d52a9046d4a81318a9163f114164269c0d71b031